### PR TITLE
feat(kit): expose `importNuxtPackage` kit utility

### LINKED ISSUE #13153: `importNuxt` uitlity
We could extract functionality in [`loader/nuxt`](https://github.com/nuxt/framework/blob/main/packages/kit/src/loader/nuxt.ts) to have a kit utility importing right nuxt version from right search path (ref https://github.com/nuxt/framework/pull/2412/files#r772270707)

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves nuxt/nuxt#13153

### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This adds a kit utility (`importNuxtPackage`) that can be used elsewhere (Bridge, probably) to import the core Nuxt package when wanting to access some other export from it (or not immediately instantiate Nuxt itself).

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/loader/nuxt.ts [hunk 1]
```diff
@@ -18,43 +18,58 @@ export interface LoadNuxtOptions extends LoadNuxtConfigOptions {
...
 
-export async function loadNuxt (opts: LoadNuxtOptions): Promise<Nuxt> {
-  // Backward compatibility
-  opts.cwd = opts.cwd || opts.rootDir
-  opts.overrides = opts.overrides || opts.config || {}
-
-  // Apply dev as config override
-  opts.overrides.dev = !!opts.dev
-
+export async function importNuxtPackage<T = any> (rootDir = process.cwd()): Promise<{ version: 2 | 3, package: T }> {
```
Comment (pi0) [inline packages/kit/src/loader/nuxt.ts:21]: We could also support `{ version: '2' | '3' }` option for both perf and making explicit nuxt version resolution

File: packages/kit/src/loader/nuxt.ts [hunk 2]
```diff
@@ -18,43 +18,58 @@ export interface LoadNuxtOptions extends LoadNuxtConfigOptions {
...
   const nearestNuxtPkg = await Promise.all(['nuxt3', 'nuxt', 'nuxt-edge']
-    .map(pkg => resolvePackageJSON(pkg, { url: opts.cwd }).catch(() => null)))
+    .map(pkg => resolvePackageJSON(pkg, { url: rootDir }).catch(() => null)))
```
Comment (pi0) [inline packages/kit/src/loader/nuxt.ts:23]: Refactor: It is actually cwd not rootDir because of node_module resolution it can also look-up

Comment (pi0) [reply packages/kit/src/loader/nuxt.ts:23]: Also we might resolove `rootDir = resolve('.', rootDir)` for consistency once in first line of function

File: packages/kit/src/loader/nuxt.ts [hunk 3]
```diff
@@ -18,43 +18,58 @@ export interface LoadNuxtOptions extends LoadNuxtConfigOptions {
...
+  rootDir = pathToFileURL(rootDir).href
 
   // Nuxt 3
   if (majorVersion === 3) {
-    const { loadNuxt } = await importModule((pkg as any)._name || pkg.name, rootDir)
-    const nuxt = await loadNuxt(opts)
-    return nuxt
+    return {
+      version: majorVersion,
+      package: await importModule((pkg as any)._name || pkg.name, rootDir)
```
Comment (pi0) [inline packages/kit/src/loader/nuxt.ts:39]: I would call it `exports`.

File: packages/kit/src/loader/nuxt.ts [hunk 4]
```diff
@@ -18,55 +19,87 @@ export interface LoadNuxtOptions extends LoadNuxtConfigOptions {
...
+    return {
+      version: 3,
+      exports: await importModule((pkg as any)._name || pkg.name, opts.cwd)
+    }
+  }
+
+  // Nuxt 2
+  return {
+    version: 2,
+    exports: await tryImportModule('nuxt-edge', opts.cwd) || await importModule('nuxt', opts.cwd)
```
[the remark this answers is not in the document]
Comment (danielroe) [reply packages/kit/src/loader/nuxt.ts:70]: nice idea

File: packages/kit/src/loader/nuxt.ts [hunk 5]
```diff
@@ -18,55 +19,87 @@ export interface LoadNuxtOptions extends LoadNuxtConfigOptions {
...
-  opts.overrides = opts.overrides || opts.config || {}
+export interface ImportNuxtPackageOptions {
+  cwd?: string
+  version?: 2 | 3
+}
 
-  // Apply dev as config override
-  opts.overrides.dev = !!opts.dev
+const nuxtPackages = {
+  2: ['nuxt-edge', 'nuxt'],
```
Comment (pi0) [inline packages/kit/src/loader/nuxt.ts:28]: Notice that `2` here is type of number while the `ImportNuxtPackageOptions["version"]` requires a number. It works with implicit type casting

File: packages/kit/src/loader/nuxt.ts [hunk 6]
```diff
@@ -18,55 +19,87 @@ export interface LoadNuxtOptions extends LoadNuxtConfigOptions {
...
+  3: ['nuxt3', 'nuxt'],
+  all: ['nuxt3', 'nuxt', 'nuxt-edge']
```
Comment (pi0) [inline packages/kit/src/loader/nuxt.ts:30]: Maybe `any` could be more semantice?

**Review by pi0 — DISMISSED** [review]

**Comment by danielroe** [comment]
closing for now - we can reimplement if needed later on

**Pull request closed without merging**